core/src/test/java/org/hibernate/ogm/backendtck/jpa/JPAConfigWithoutProviderTest.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.backendtck.jpa;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.EntityTransaction;
+import javax.persistence.Persistence;
+import javax.persistence.spi.PersistenceProvider;
+import javax.persistence.spi.PersistenceProviderResolver;
+import javax.persistence.spi.PersistenceProviderResolverHolder;
+
+import org.hibernate.jpa.HibernatePersistenceProvider;
+import org.hibernate.ogm.cfg.OgmProperties;
+import org.hibernate.ogm.jpa.HibernateOgmPersistence;
+import org.hibernate.ogm.utils.PackagingRule;
+import org.hibernate.ogm.utils.TestForIssue;
+import org.hibernate.ogm.utils.TestHelper;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+/**
+ * Tests a JPA configuration without defining a persistence provider on it.
+ *
+ * The order of resolution of the {@link PersistenceProvider}s is unpredictable a priori,
+ * then we have to repeat the same test over all possible dispositions.
+ * It has to work in any cases.
+ *
+ * @author Fabio Massimo Ercoli
+ */
+@TestForIssue(jiraKey = "OGM-1517")
+public class JPAConfigWithoutProviderTest {
+
+	private static final String POEM_NAME = "The Orlando Furioso";
+
+	@Rule
+	public PackagingRule packaging = new PackagingRule( "persistencexml/ogm-noprovider.xml", Poem.class );
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	private EntityManagerFactory factory;
+	private EntityManager manager;
+
+	@Test
+	public void testNoProviderConfigWhenORMProviderBeforeOGMProvider() {
+		forceORMProviderBeforeOGMProvider();
+		testNoProviderConfig();
+	}
+
+	@Test
+	public void testNoProviderConfigWhenOGMProviderIsDeclaredBeforeORMProvider() {
+		forceOGMProviderBeforeORMProvider();
+		testNoProviderConfig();
+	}
+
+	@Test
+	public void testNoProviderConfigWithDefaultProviderResolver() {
+		restoreDefaultProviderResolver();
+		testNoProviderConfig();
+	}
+
+	@Test
+	public void testNoProviderConfigWithORMProviderOnly() {
+		forceORMProviderOnly();
+		testNoProviderConfig();
+	}
+
+	@Test
+	public void testNoProviderConfigWithORMProviderOnlyUsingEnableProperty() {
+		forceORMProviderOnly();
+		testNoProviderConfigUsingEnableProperty();
```

Comment:
I appreciate the sentiment behind avoiding duplication but this approach  makes it harder to understand what each test is doing. It also doesn't work very well if at some point we want to assert something different in one of the tests.

I think these tests would be more readable if you write them in the form:
```
@Test
testExample() {
    settings = defaultSettings();
    emf = createEMF( "no-properties-pu", settings ); // Do we also need to check the EM?
    assertThat( emf ).isInstanceOf( ... ); 
    assertEntityCRUD( .emf ); // Optional. seems unrelated but I don't mind.
}
```

I wouldn't test with CRUD operations because we already have all the rest of the suit for that. It seems overkill to me.
But if you want to add it keep it in the test method so that someone else can figure out what's going on at first glance.

Reply:
I agree. I'll to change it